From b29ac30b36ce72796c60328adc93d73c8902fa84 Mon Sep 17 00:00:00 2001
From: Ratheesh Kannoth <rkannoth@codeaurora.org>
Date: Fri, 27 Nov 2020 07:16:55 +0530
Subject: [PATCH] net: openvswitch: Delay flow delete notification

Notify flow delete event after removing from datapath flow table

Change-Id: Ib80da37088b411831d63325b688c64aed7dfecdb
Signed-off-by: Pavithra R <quic_pavir@quicinc.com>
---
 net/openvswitch/datapath.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/net/openvswitch/datapath.c
+++ b/net/openvswitch/datapath.c
@@ -1519,8 +1519,8 @@ static int ovs_flow_cmd_del(struct sk_bu
 		goto unlock;
 	}
 
-	ovs_dp_flow_del_notify(dp, flow);
 	ovs_flow_tbl_remove(&dp->table, flow);
+	ovs_dp_flow_del_notify(dp, flow);
 	ovs_unlock();
 
 	reply = ovs_flow_cmd_alloc_info((const struct sw_flow_actions __force *) flow->sf_acts,
